common/djangoapps/student/tests/test_views.py

```diff
@@ -243,3 +282,27 @@ def test_redirect_account_settings(self):
         UserProfile.objects.get(user=self.user).delete()
         response = self.client.get(self.path)
         self.assertRedirects(response, reverse('account_settings'))
+
+    @patch.multiple('django.conf.settings', **MOCK_SETTINGS)
+    @ddt.data(
+        *itertools.product(
+            [TOMORROW],
+            [True, False],
+            [True, False],
+        )
+    )
+    @ddt.unpack
+    def test_sharing_icons_for_future_course(self, start_date, set_marketing, set_social_sharing):
+        """
+        Verify that the course sharing icons show up if course is starting in future and
+        any of of marketing or social sharing url is set.
+        """
+        self.course = CourseFactory.create(start=start_date, emit_signals=True)
+        self.course_enrollment = CourseEnrollmentFactory(course_id=self.course.id, user=self.user)
+
+        self.set_course_sharing_urls(set_marketing, set_social_sharing)
+
+        # Assert course sharing icons
+        response = self.client.get(reverse('dashboard'))
+        self.assertEqual('Share on Twitter' in response.content, set_marketing or set_social_sharing)
+        self.assertEqual('Share on Facebook' in response.content, set_marketing or set_social_sharing)
```

Comment:
I'd like to add a test where we won't be showing sharing links i.e. where we only have course about link.